pkg/operator/baremetal_config.go

```diff
@@ -0,0 +1,57 @@
+package operator
+
+import (
+	"github.com/golang/glog"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	dynamic "k8s.io/client-go/dynamic"
+)
+
+var provisioningGVR = schema.GroupVersionResource{Group: "metal3.io", Resource: "provisionings", Version: "v1alpha1"}
+
+func getBaremetalProvisioningConfig(dc dynamic.Interface, configName string) BaremetalConfig {
+	provisioningClient := dc.Resource(provisioningGVR)
+	provisioningConfig, err := provisioningClient.Get(configName, metav1.GetOptions{})
+	if err != nil {
+		glog.V(1).Infof("Error getting provisioning config %q", configName)
+		return BaremetalConfig{}
+	}
+	provisioningSpec, found, err := unstructured.NestedMap(provisioningConfig.UnstructuredContent(), "spec")
+	if !found {
+		glog.Errorf("Nested Spec not found in Baremetal provisioning CR %q", configName)
+		return BaremetalConfig{}
+	}
+	provisioningInterface, found, err := unstructured.NestedString(provisioningSpec, "provisioningInterface")
+	if !found {
+		glog.Errorf("provisioningInterface not found in Baremetal provisioning CR %q", configName)
+		return BaremetalConfig{}
+	}
+	provisioningIP, found, err := unstructured.NestedString(provisioningSpec, "provisioningIP")
+	if !found {
+		glog.Errorf("provisioningIP not found in Baremetal provisioning CR %q", configName)
+		return BaremetalConfig{}
+	}
+	provisioningNetworkCIDR, found, err := unstructured.NestedString(provisioningSpec, "provisioningNetworkCIDR")
+	if !found {
+		glog.Errorf("provisioningNetworkCIDR not found in Baremetal provisioning CR %q", configName)
+		return BaremetalConfig{}
+	}
+	provisioningDHCPExternal, found, err := unstructured.NestedBool(provisioningSpec, "provisioningDHCPExternal")
+	if !found {
+		glog.Errorf("provisioningDHCPExternal not found in Baremetal provisioning CR %q", configName)
+		return BaremetalConfig{}
+	}
+	provisioningDHCPRange, found, err := unstructured.NestedString(provisioningSpec, "provisioningDHCPRange")
+	if !found {
+		glog.Errorf("provisioningDHCPRange not found in Baremetal provisioning CR %q", configName)
+		return BaremetalConfig{}
+	}
+	return BaremetalConfig{
```

Comment:
wouldn't `BaremetalProvisioningConfig` be more appropriate?

Reply:
Done.